### DISCUSSION TITLE
`CancelToken` for long running queries

### DISCUSSION THREAD

**Discussion by naim94a** [weiznich/diesel_async#57] [Q&A]
Hi!,
I'm new to diesel, it looks awesome.
I would like to use diesel in a project where queries may take a long time to complete, but if the client disconnects - I would like to cancel the query. AFAIK, tokio-postgres won't signal the server to cancel a query if the future is dropped. `tokio-postgres` provides a token that can be used to attempt to terminate an ongoing query via the [`cancel_token`](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Client.html#method.cancel_token) function.
Is it possible to access `cancel_token` somehow before executing a query?

Thanks!

**Comment by weiznich** [comment]
This functionality is currently not exposed by diesel-async. I would accept a PR adding this. It should include documentation and a small test.

**Reply by naim94a** [reply]
Cool, I have submitted PR #58.
